src/hoc/withValidation.js

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { setValidationParam, getOrderValidation, setIsTouched } from '../redux/validationRedux';
+
+// eslint-disable-next-line import/no-anonymous-default-export
+export default WrappedComponent => {
+
+  const ValidationHOC = ({ className, ...props }) => {
+    const { orderValidation, setValidationParam, setIsTouched } = props;
+    const [ formData, setFormData ] = useState('');
+
+    let updateAction;
+    const getUpdateAction = action => updateAction = action;
+
+    useEffect(() => {
+      updateAction(formData);
+    }, [updateAction, formData]);
+
+    const ruleFunctions = {
+      required: text => text && text.length > 0,
+      pattern: (text, pattern) => pattern && pattern.test(text),
+      minLength: (text, minLength) => text && text.length ? text.length >= minLength : true,
+      maxLength: (text, maxLength) => text && text.length ?  text.length <= maxLength : true,
+    };
+
+    const checkValidation = (form, field, inputText, rules) => {
+      Object.keys(rules).forEach(rule => {
```

Comment:
Jeśli nazwy metod oraz nazwy wymagań validacyjnych są takie same, to druga pętla jest niepotrzebna.
Object.keys(rules).forEach(rule => {
  ruleFunctions[rule]
})

Ewentualnie z dodatkowym sprawdzeniem czy reguła faktycznie istnieje, typeof ruleFunctions[rule] !== 'undefined'